src/terms/termLib.ml

```diff
@@ -256,19 +256,17 @@ type logic = [ `None | `Inferred of features | `SMTLogic of string ]
 
 let pp_print_logic fmt = function
   | `None -> pp_print_string fmt "ALL"
-  | `Inferred l -> if (L.mem BV l) then 
-                    pp_print_string fmt "ALL"
-                   else 
-                    pp_print_features fmt l
+  | `Inferred l ->
+      if L.mem BV l && (L.mem IA l || L.mem RA l) then pp_print_string fmt "ALL"
+      else pp_print_features fmt l
   | `SMTLogic s -> pp_print_string fmt (if s = "" then "ALL" else s)
 
 
 let string_of_logic = function
   | `None -> "ALL"
-  | `Inferred l -> if (L.mem BV l) then 
-                    "ALL"
-                   else
-                    string_of_features l
+  | `Inferred l ->
+      if L.mem BV l && (L.mem IA l || L.mem RA l) then "ALL"
+      else string_of_features l
   | `SMTLogic s -> if s = "" then "ALL" else s
```

Comment:
Define this function in terms of `pp_print_logic` to synchronize logic:
`let string_of_logic l = asprintf "%a" pp_print_logic l`

Reply:
Done!